hive-metastore/src/main/java/org/apache/iceberg/hive/legacy/HiveExpressions.java

```diff
@@ -48,9 +49,17 @@ private HiveExpressions() {}
    */
   static Expression simplifyPartitionFilter(Expression expr, Set<String> partitionColumnNames) {
     try {
-      Expression partitionPredicatesOnly = ExpressionVisitors.visit(expr,
+      // Pushing down NOTs is critical for the correctness of RemoveNonPartitionPredicates
+      // Without pushdown NOT(P and NP) will be written to NOT(P)
```

Comment:
With simply ignoring NP, ..
Also may clarify what P and NP are

Reply:
Done